src/components/UserInfo.tsx

```diff
@@ -0,0 +1,16 @@
+import { User } from '../types/types';
+
+type Props = {
+  user: User | null,
+};
+
+export const UserInfo: React.FC<Props> = ({ user }) => {
+  return (
+    user && (
```

Comment:
check it in a parent component, not need to render UserInfo at all if user is null